base/reducedim.jl

```diff
@@ -343,13 +375,45 @@ prod!(r, A)
 Compute the maximum value of an array over the given dimensions. See also the
 [`max(a,b)`](@ref) function to take the maximum of two or more arguments,
 which can be applied elementwise to arrays via `max.(a,b)`.
+
+```jldoctest
+julia> A = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+
+julia> maximum(A, 1)
+1×2 Array{Int64,2}:
+ 3  4
+ 
+julia> maximum(A, 2)
+2×1 Array{Int64,2}:
+ 2
+ 4
+```
 """
 maximum(A, dims)
 
 """
     maximum!(r, A)
 
 Compute the maximum value of `A` over the singleton dimensions of `r`, and write results to `r`.
+
+```jldoctest
+julia> A = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+
+julia> maximum!([1; 1], A)
```

Comment:
using a matrix for the dimensions argument is a little funny and could maybe be an error?

(or a keyword could be used for dims arguments but that would be a big change)